LFPy/eegmegcalc.py

```diff
@@ -0,0 +1,476 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jan 15 17:24:00 2015
+
+@author: solveig
+"""
+
+from __future__ import division
+from scipy.special import eval_legendre, lpmv
+import numpy as np
+
+class EEGMEGCalc:
+    """ Import all data needed for calculating the extracellular potential Phi at location r
+    from a current dipole moment p at location rz. Also decompose p into a radial and a tangential part.
+    """
+
+    def __init__(self, radii, sigmas, r, rz):
+
+        self.k1 = 1E6  # from mV to nV
+        self.r1 = radii[0]
+        self.r2 = radii[1]
+        self.r3 = radii[2]
+        self.r4 = radii[3]
+
+        self.sigma1 = sigmas[0]
+        self.sigma2 = sigmas[1]
+        self.sigma3 = sigmas[2]
+        self.sigma4 = sigmas[3]
+
+        self.r12 = self.r1 / self.r2
+        self.r21 = self.r2 / self.r1
+        self.r23 = self.r2 / self.r3
+        self.r32 = self.r3 / self.r2
+        self.r34 = self.r3 / self.r4
+        self.r43 = self.r4 / self.r3
+
+        self.sigma12 = self.sigma1 / self.sigma2
+        self.sigma21 = self.sigma2 / self.sigma1
+        self.sigma23 = self.sigma2 / self.sigma3
+        self.sigma32 = self.sigma3 / self.sigma2
+        self.sigma34 = self.sigma3 / self.sigma4
+        self.sigma43 = self.sigma4 / self.sigma3
+
+        self.rxyz = r
+        self.rzloc = rz
+        self.rz = np.sqrt(np.sum(rz ** 2))
+        self.rz1 = self.rz / self.r1
+        self.r = np.sqrt(np.sum(r ** 2, axis=1))
+
+        # if self.r < self.rz:
+        #     raise ValueError('Electrode must be farther away from brain center than dipole. r = %s, rz = %s', self.r, self.rz)
+        # elif self.r > self.r4:
+        #     raise ValueError('Electrode located outside head model. Your r = %s µm. Maximum r = %s µm.', self.r, self.r4)
```

Comment:
I think such commented out code can be removed (unless you intend to put the code back).